tests/snapshots/test_metric_filter_rendering.py/SqlQueryPlan/DuckDB/test_query_with_cumulative_metric_in_where_filter__plan0_optimized.sql

```diff
@@ -0,0 +1,39 @@
+-- Constrain Output with WHERE
+-- Pass Only Elements: ['listings',]
+-- Aggregate Measures
+-- Compute Metrics via Expressions
+SELECT
+  SUM(listings) AS listings
+FROM (
+  -- Join Standard Outputs
+  -- Pass Only Elements: ['listings', 'user__revenue_all_time']
+  SELECT
+    subq_26.revenue_all_time AS user__revenue_all_time
+    , subq_20.listings AS listings
+  FROM (
+    -- Read Elements From Semantic Model 'listings_latest'
+    -- Metric Time Dimension 'ds'
+    -- Pass Only Elements: ['listings', 'user']
+    SELECT
+      user_id AS user
+      , 1 AS listings
+    FROM ***************************.dim_listings_latest listings_latest_src_28000
+  ) subq_20
+  LEFT OUTER JOIN (
+    -- Read Elements From Semantic Model 'revenue'
+    -- Metric Time Dimension 'ds'
+    -- Pass Only Elements: ['txn_revenue', 'user']
+    -- Aggregate Measures
+    -- Compute Metrics via Expressions
+    -- Pass Only Elements: ['user', 'revenue_all_time']
+    SELECT
+      user_id AS user
+      , SUM(revenue) AS revenue_all_time
```

Comment:
Yes! I didn't add any with `window` or `grain_to_date` yet because those require querying with `metric_time`, which is not yet supported for metric group bys. So you'll get an error if you try to use them anyway.